chall06/fulldemo.c

```diff
@@ -0,0 +1,259 @@
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+int board_check(char *board)
+{
+	int len;
+	int i;
+	int king;
+	int count;
+
+	i  = 0;
+	king = 0;
+	count = 0;
+	while (board[i] != '\0')
+	{
+		if (board[i] != '\n' && board[i] != 'K' && board[i] != 'P'
+		&& board[i] != 'B' && board[i] != 'R' && board[i] != 'Q' 
+		&& board[i] != '.')
+			return (0);
+		if (board[i] == 'K')
+			king++;
+		if (board[i] == 'P' || board[i] == 'B' || board[i] == 'R' || board[i] == 'Q')
+			count++;
+		i++;
+	}
+	if (king > 1 || king == 0 || count == 0)
+		return (0);
+	len = 0;
+	while (board[len] != '\n' && board[len] != '\0')
+		len++;
+	return (len);
+}
+
+int clear_board(char *str, char **board, int r)
+{
+	int len;
+	int i;
+
+	i = 0;
+	if (board)
+	{
+		len = strlen(board[0]);
+		while (i < len)
+		{
+			free(board[i]);
+			i++;
+		}
+		free(board);
+	}
+	if (str)
+		free(str);
+	return (r);
+}
+
+char **board_creator(char *board, int len)
+{
+	char **map;
+	char *row;
+	int i;
+	int length;
+	int k;
+
+	i  = 0;
+	if (len > 9 || !(map = (char **)malloc(sizeof(char *) * len)))
+		return (NULL);
+	row = strtok(board, "\n");
+	length = strlen(row);
+	while (i < len)
+	{
+		if (!(map[i] = strdup(row)))
+			return (NULL);
+		row = strtok(NULL, "\n");
+		if (row)
+			if (strlen(row) != length)
+				return (NULL);
+		i++;
+	}
+	return (map);
+}
+int rook(char **map, int i, int j)
+{
+	int n;
+	int len;
+
+	len = strlen(map[0]) - 1;
```

Comment:
Because of this `- 1` you are missing all the pieces that are on the right and bottom extremities of the board, e.g:
```
RK
..
```